src/arch/mod.rs

```diff
@@ -31,21 +31,27 @@ lazy_static! {
 mod x64;
 pub use x64::*;
 
-pub fn union_size(symbols: &[Symbol]) -> Result<SIZE_T, &'static str> {
+/// Calculate the size of a union.
```

Comment:
```suggestion
/// Calculate the size of a union: the max of all member sizes
```

Also, this reminds me that I need to round this to the largest alignment. This is not necessarily a divisor of the largest size! Take for instance `union { long i; char a[3]; }`: the size is 6, but the alignment is 4.